From a63efc5b61b1659f015481fc6db6275db96daf66 Mon Sep 17 00:00:00 2001
From: Yao Zi <ziyao@disroot.org>
Date: Wed, 19 Feb 2025 05:30:01 +0000
Subject: [PATCH] amdgpu: dml2: Increase max stack size

AMDGPU dml2 fails to build with Clang 19 with error like

error: stack frame size (2136) exceeds limit (2048) in 'dml_core_mode_support' [-Werror,-Wframe-larger-than]

on x86_64. There has been report that AMDGPU exceeds stack size
limitation when KASAN/KCSAN is on. But we encounter similar problem
even with these options set to off.

Let's increase the allowed stack size as a workaround.

Signed-off-by: Yao Zi <ziyao@disroot.org>
---
 drivers/gpu/drm/amd/display/dc/dml2/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/display/dc/dml2/Makefile b/drivers/gpu/drm/amd/display/dc/dml2/Makefile
index 986a69c..d78aac1 100644
--- a/drivers/gpu/drm/amd/display/dc/dml2/Makefile
+++ b/drivers/gpu/drm/amd/display/dc/dml2/Makefile
@@ -35,7 +35,7 @@ else
 frame_warn_flag := -Wframe-larger-than=3072
 endif
 else
-frame_warn_flag := -Wframe-larger-than=2048
+frame_warn_flag := -Wframe-larger-than=3072
 endif
 endif
 
-- 
2.48.1

